internal/batches/service/service.go

```diff
@@ -380,9 +381,43 @@ func (svc *Service) ParseBatchSpec(data []byte) (*batcheslib.BatchSpec, error) {
 	if err != nil {
 		return nil, errors.Wrap(err, "parsing batch spec")
 	}
+	if err = handleMount(dir, spec); err != nil {
+		return nil, errors.Wrap(err, "parsing batch spec")
+	}
 	return spec, nil
 }
 
+func handleMount(dir string, spec *batcheslib.BatchSpec) error {
+	for i, step := range spec.Steps {
+		for j, mount := range step.Mount {
+			p := mount.Path
+			if !filepath.IsAbs(p) {
+				// Try to build the absolute path since Docker will only mount absolute paths
+				p = filepath.Join(dir, p)
+			}
+			pathInfo, err := os.Stat(p)
+			if os.IsNotExist(err) {
+				return errors.Newf("step %d mount path %s does not exist", i+1, p)
+			} else if err != nil {
+				return errors.Wrapf(err, "step %d mount path validation", i+1)
+			} else {
+				if !strings.HasPrefix(p, dir) {
+					return errors.Newf("step %d mount path is not in the same directory or subdirectory as the batch spec", i+1)
+				}
+				// Mounting a directory on Docker must end with the separator. So, append the file separator to make
+				// users' lives easier.
+				if pathInfo.IsDir() && !strings.HasSuffix(p, string(filepath.Separator)) {
+					p += string(filepath.Separator)
+				}
+				// Update the mount path to the absolute path so building the absolute path (above) does not need to be
+				// redone when adding the mount argument to the Docker container.
+				step.Mount[j].Path = p
```

Comment:
I don't see any issues with caching, but I am not super familiar with the cache yet. 

Actually, I was having issues with the cache where I would only make a change to the script file and would rerun the batch spec and nothing would change because of the cache (for example changing the text to append to the `README.md` from `Hello from python` to `Hello from python1`).

I was getting slightly annoyed with this issue, and I am sure another user will too. Is this something we want to deal with in this Issue or in a follow up Issue?

Reply:
I just saw the flag `-clear-cache`. So I guess, it is really not a problem? Maybe we should have some docs around the mount and cache and communicate that the `-clear-cache` flag is available when updating the script[s]?